src/createScope.tsx

```diff
@@ -18,60 +18,69 @@ export const ScopeProvider = ({
   atoms: Iterable<AnyAtom>;
   children: ReactNode;
 }) => {
-  const [getParentScopedAtom] = useContext(ScopeContext);
-  const mapping = new WeakMap<AnyAtom, AnyAtom>();
-  const atomSet = new Set(atoms);
+  const [getParentScopedAtom, parentScopedAtoms] = useContext(ScopeContext);
+  const getScopedAtomAndScopedAtoms = useMemo(() => {
+    const mapping = new WeakMap<AnyAtom, AnyAtom>();
+    const atomSet = new Set(atoms);
+    const scopedAtoms = new Set([...parentScopedAtoms, ...atoms] as AnyAtom[]);
```

Comment:
`scopedAtoms` is a cache containing all the scoped atoms. Therefore, if an atom is not scoped, it does not have to call `getParentScopedAtom`/`createScopedAtom`.